include/wpe/process.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2022 Igalia S.L.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !defined(__WPE_H_INSIDE__) && !defined(WPE_COMPILATION)
+#error "Only <wpe/wpe.h> can be included directly."
+#endif
+
+#ifndef wpe_process_h
+#define wpe_process_h
+
+/**
+ * SECTION:process
+ * @short_description: Process management
+ * @title: Process
+ */
+
+#if defined(WPE_ENABLE_PROCESS) && WPE_ENABLE_PROCESS
+
+#if defined(WPE_COMPILATION)
+#include "export.h"
+#endif
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum wpe_process_type {
+    WPE_PROCESS_TYPE_WEB,
+    WPE_PROCESS_TYPE_NETWORK,
+    WPE_PROCESS_TYPE_GPU,
+    WPE_PROCESS_TYPE_WEB_AUTHN,
+};
+
+struct wpe_process_provider;
+
+struct wpe_process_provider_interface {
+    void* (*create)(struct wpe_process_provider*);
+    void (*destroy)(void*);
+    int32_t (*launch)(void*, enum wpe_process_type, void*);
+    void (*terminate)(void*, int32_t);
+};
+
+WPE_EXPORT
+struct wpe_process_provider* wpe_process_provider_create(void);
+
+WPE_EXPORT
+void wpe_process_provider_destroy(struct wpe_process_provider*);
+
+WPE_EXPORT
+int32_t wpe_process_launch(struct wpe_process_provider*, enum wpe_process_type, void*);
```

Comment:
ProcessLauncherGLib webkit uses uint64 type for process identifier. Maybe the return value should be uint64_t if it's process identifier

Reply:
Currently WTF defines `ProcessID` to be `int` on Windows and `pid_t` on the rest of systems. According to POSIX the requirement is that `pid_t` is a signed integral type of some system-dependent size, so the best here would be to go for the biggest integer type we can expect all platforms to supported.

I would use `int64_t` as suggested by @zhani, or even `intmax_t` if we want to do maximum future-proofing. Either way would be fine by me.

Reply:
`int64_t` or `uint64_t`?

Reply:
I would go with `int64_t` (signed) for consistency with WTF. If some system needs it to be unsigned, casting values is always a possibility.